cluster/src/main/java/org/apache/iotdb/cluster/query/ClusterPlanExecutor.java

```diff
@@ -707,5 +709,24 @@ public void delete(Path path, long timestamp) throws QueryProcessException {
     }
   }
 
+  @Override
+  public void delete(DeletePlan deletePlan) throws QueryProcessException {
+    try {
+      Set<String> existingPaths = new HashSet<>();
+      for (Path p : deletePlan.getPaths()) {
+        existingPaths.addAll(getPathsName(p.getFullPath()));
```

Comment:
`getFullPath()` here will still perform a metadata query on other nodes, so I doubt whether your changes work.

Reply:
These codes seems redundent now, because the DeletePlan to be executed contains full paths. I will remove these codes.